scripts/notebooks/SampleQC.ipynb

```diff
@@ -0,0 +1,2819 @@
+    "# )\n",
+    "\n",
+    "# with open('sample_set.zip', 'wb') as f:\n",
+    "#     f.write(sample_set_response.content)\n",
+    "    \n",
+    "# with zipfile.ZipFile('sample_set.zip', 'r') as zip_ref:\n",
+    "#     # Extract sample set data\n",
+    "#     with zip_ref.open('sample_set_entity.tsv') as file:\n",
+    "#         tsv_file = io.StringIO(file.read().decode('utf-8'))\n",
+    "#         sample_set_tbl = pd.read_csv(tsv_file, sep='\\t')\n",
+    "#         sample_set_tbl = sample_set_tbl[sample_set_tbl['ploidy_plots'].notnull() & sample_set_tbl['qc_table'].notnull()]\n",
+    "#         sample_set_tbl = sample_set_tbl.reset_index(drop=True)\n",
+    "    \n",
+    "#     # Extract sample membership data\n",
+    "#     with zip_ref.open('sample_set_membership.tsv') as membership_file:\n",
+    "#         membership_tsv = io.StringIO(membership_file.read().decode('utf-8'))\n",
+    "#         membership_df = pd.read_csv(membership_tsv, sep='\\t')\n",
+    "    \n",
+    "#     # Add list of samples to corresponding sample set\n",
+    "#     sample_groups = membership_df.groupby('membership:sample_set_id')['sample'].unique().apply(list)\n",
+    "#     sample_set_tbl['samples'] = sample_set_tbl['entity:sample_set_id'].map(sample_groups)\n",
+    "#     sample_set_tbl['samples'] = sample_set_tbl['samples'].apply(lambda x: x if isinstance(x, list) else [])\n",
+    "\n",
+    "# print(\"Batches:\")\n",
+    "# print(*list(sample_set_tbl['entity:sample_set_id'].values), sep='\\n')\n",
+    "\n",
+    "# file_path = generate_file_path(TLD_PATH, 'artifacts', 'sample_sets_qc.tsv')\n",
+    "# save_df(WS_BUCKET, file_path, sample_set_tbl)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Use this block to reload the data saved in the previous cell\n",
+    "file_path = generate_file_path(TLD_PATH, 'artifacts', 'sample_sets_qc.tsv')\n",
+    "\n",
+    "sample_set_tbl = pd.read_table(file_path, sep='\\t', dtype={'sample_id': str})\n",
+    "\n",
+    "sample_set_tbl"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<div class=\"alert alert-block alert-success\"><b>Optional</b>: If you wish to only include a subset of batches out of the ones listed above, you can filter <tt>sample_set_tbl</tt> to only include them using the cell below. Ensure that all batches you expect are included.</div>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Optionally filter to only include a subset of batches from sample_set_tbl and update the saved file\n",
+    "\n",
+    "\n",
+    "print(\"Batches:\")\n",
+    "print(*list(sample_set_tbl['entity:sample_set_id'].values), sep='\\n')\n",
+    "\n",
+    "file_path = generate_file_path(TLD_PATH, 'artifacts', 'sample_sets_qc.tsv')\n",
+    "save_df(WS_BUCKET, file_path, sample_set_tbl)"
```

Comment:
Updated accordingly - redundant if moved to next cell (since we already print this post-filtering) so just removing all together. Users can see their current batches as we display already `sample_set_tbl` before filtering, so I guess we don't need to explicitly print batches again.